frontend/src/AlertsManagement/AlertInfo.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component, Fragment } from 'react';
+import { Line } from 'react-chartjs-2';
+import { Link } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+import Divider from '@material-ui/core/Divider';
+import IconButton from '@material-ui/core/IconButton';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import Typography from '@material-ui/core/Typography';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import { getAlertVisData } from './management_helpers';
+import { UNRESOLVED_STATUS, RESOLVED_STATUS } from './management_constants';
+
+/**
+ * Requests alert data given specified alert ID from route and 
+ * visualizes the historical metric data using a chart.js wrapper for React.
+ */
+class AlertInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alert: null,
+    }
+  }
+
+  async componentDidMount() {
+    var result = await getAlertVisData(this.props.match.params.alertId);
+    // TODO: instead of checking for null response, check for 404 status code.
+    if (result === null) {
+      throw new Error("Could not find alert.")
+    }
+    this.setState({ alert: result });
+  }
+
+  handleStatusChange = () => {
```

Comment:
is this a synchronous or asynchronous fn?

Reply:
I think it's synchronous. The `setState` inside the function is asynchronous but because I'm updating the frontend state and sending the POST request, I don't have to wait for the POST request to finish (the POST request is mainly for maintaining property in the Datastore).